From 757a759d5ef4fac4d719998c68c3ebe3e6ad5998 Mon Sep 17 00:00:00 2001
From: jqdeng <Emily.Deng@amd.com>
Date: Tue, 5 Jul 2016 15:45:33 +0800
Subject: [PATCH 18/39] amdgpu/tests: Add the test case for amdgpu_get_fb_id
 and amdgpu_get_bo_from_fb_id v2

v2:
502c3f21127b5253c46ae23bdd563dbce4fb57ae
[Flora Cui]
stop fb_id test for ASIC without output

Signed-off-by: jqdeng <Emily.Deng@amd.com>
Reviewed-by: Chunming Zhou <David1.Zhou@amd.com>
Signed-off-by: Flora Cui <Flora.Cui@amd.com>
Signed-off-by: Arsalan H. Awan <Arsalan_Awan@mentor.com>
---
 tests/amdgpu/bo_tests.c | 21 +++++++++++++++++++++
 1 file changed, 21 insertions(+)

diff --git a/tests/amdgpu/bo_tests.c b/tests/amdgpu/bo_tests.c
index 9d4da4a..2ab31c1 100644
--- a/tests/amdgpu/bo_tests.c
+++ b/tests/amdgpu/bo_tests.c
@@ -44,6 +44,7 @@ static void amdgpu_bo_metadata(void);
 static void amdgpu_bo_map_unmap(void);
 static void amdgpu_memory_alloc(void);
 static void amdgpu_mem_fail_alloc(void);
+static void amdgpu_get_fb_id_and_handle(void);
 
 CU_TestInfo bo_tests[] = {
 	{ "Export/Import",  amdgpu_bo_export_import },
@@ -51,6 +52,7 @@ CU_TestInfo bo_tests[] = {
 	{ "CPU map/unmap",  amdgpu_bo_map_unmap },
 	{ "Memory alloc Test",  amdgpu_memory_alloc },
 	{ "Memory fail alloc Test",  amdgpu_mem_fail_alloc },
+	{ "GET FB_ID AND FB_HANDLE",  amdgpu_get_fb_id_and_handle },
 	CU_TEST_INFO_NULL,
 };
 
@@ -194,6 +196,25 @@ static void amdgpu_bo_map_unmap(void)
 	CU_ASSERT_EQUAL(r, 0);
 }
 
+static void amdgpu_get_fb_id_and_handle(void)
+{
+	uint32_t *ptr;
+	int i, r;
+	unsigned int fb_id;
+	struct amdgpu_bo_import_result output;
+
+	r = amdgpu_get_fb_id(device_handle, &fb_id);
+	CU_ASSERT_EQUAL(r, 0);
+	if (fb_id == 0) {
+		fprintf(stderr, "\nSkipping amdgpu_get_fb_id_and_handle test: no monitor connected\n");
+		return;
+	}
+
+	r = amdgpu_get_bo_from_fb_id(device_handle, fb_id, &output);
+	CU_ASSERT_EQUAL(r, 0);
+	CU_ASSERT_NOT_EQUAL(output.buf_handle, 0);
+}
+
 static void amdgpu_memory_alloc(void)
 {
 	amdgpu_bo_handle bo;
-- 
2.7.4
